engine/src/main/java/org/destinationsol/assets/audio/OggMusicManager.java

```diff
@@ -145,53 +148,47 @@ public void changeVolume(GameOptions options) {
     /**
      * Registers all music from the module of shipName
      *
-     * @param shipName Name of the ship that is loaded when creating a new game
+     * @param moduleName Name of the module of ship that is loaded when creating a new game
      */
-    public void registerModuleMusic(String shipName) {
-        Set<ResourceUrn> configUrnList = Assets.getAssetHelper().list(Json.class, "[a-zA-Z]*:playerSpawnConfig");
-
-        for (ResourceUrn configUrn : configUrnList) {
-            Json json = Assets.getJson(configUrn.toString());
-            JSONObject rootNode = json.getJsonValue();
-
-            if (rootNode.keySet().contains(shipName) && rootNode.get(shipName) instanceof JSONObject) {
-                String moduleName = configUrn.toString().split(":")[0];
-                Json musicJson;
-
-                try {
-                    musicJson = Assets.getJson(moduleName + ":musicConfig");
-                } catch (RuntimeException e) {
-                    logger.warn("Music config not found for module " + moduleName + ":musicConfig");
-                    return;
-                }
-
-                MusicConfig musicConfig = MusicConfig.load(moduleName, musicJson.getJsonValue());
-                Map<String, List<String>> musicMap = musicConfig.getMusicMap();
-
-                this.musicMap.remove(MENU_MUSIC_SET);
-                for (String music : musicMap.get(MENU_MUSIC_SET)) {
-                    registerMusic(MENU_MUSIC_SET, music);
-                }
-                for (String music : musicMap.get(GAME_MUSIC_SET)) {
-                    registerMusic(GAME_MUSIC_SET, music);
-                    logger.info("Registered " + music);
-                }
+    public void registerModuleMusic(String moduleName, final GameOptions options) {
+        if (currentlyRegisteredModule != null) {
+            if (moduleName.equals(currentlyRegisteredModule)) {
+                //Skips directly to module music
+                stopMusic();
+                playMusicTrack(this.musicMap.get(GAME_MUSIC_SET).get(2), options);
+                return;
+            } else {
+                unregisterModuleMusic();
+                currentlyRegisteredModule = null;
             }
+        }
 
-            json.dispose();
+        if (Assets.getAssetHelper().list(Json.class, moduleName + ":musicConfig").isEmpty()) {
+            logger.warn("Music config not found for module " + moduleName + ":musicConfig");
+            return;
         }
 
+        Json musicJson = Assets.getJson(moduleName + ":musicConfig");
 
-        registerMusic(GAME_MUSIC_SET, "engine:cimmerianDawn");
-        registerMusic(GAME_MUSIC_SET, "engine:intoTheDark");
-        registerMusic(GAME_MUSIC_SET, "engine:spaceTheatre");
+        MusicConfig musicConfig = MusicConfig.load(moduleName, musicJson.getJsonValue());
+        Map<String, List<String>> musicMap = musicConfig.getMusicMap();
```

Comment:
This local `musicMap` variable _hides_ the `musicMap` instance variable declared at the top of a class, a bad practice and common cause of bugs. Your IDE should be highlighting this as something to look into. You could simply rename the local variable here but watch out - is there a bug? _Which_ `musicMap` are you needing to interact with in this method?